x/stakeibc/keeper/invariants.go

```diff
@@ -3,18 +3,122 @@ package keeper
 // DONTCOVER
 
 import (
+	"fmt"
+
 	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	// epochtypes "github.com/Stride-Labs/stride/v4/x/epochs/types"
+	"github.com/Stride-Labs/stride/v5/x/stakeibc/types"
+)
+
+const (
+	balanceStakeHostZoneInvariantName = "balance-stake-hostzone"
+	validatorWeightHostZoneInvariantName = "validator-weight-hostzone"
+	redemptionRateInvariantName = "redemption-rate"
 )
 
 // RegisterInvariants registers all governance invariants.
 func RegisterInvariants(ir sdk.InvariantRegistry, k Keeper) {
+	ir.RegisterRoute(types.ModuleName, balanceStakeHostZoneInvariantName, k.BalanceStakeHostZoneInvariant())
+	ir.RegisterRoute(types.ModuleName, validatorWeightHostZoneInvariantName, k.ValidatorWeightHostZoneInvariant())
+	ir.RegisterRoute(types.ModuleName, redemptionRateInvariantName, k.RedemptionRateInvariant())
 }
 
 // AllInvariants runs all invariants of the stakeibc module
 func AllInvariants(k Keeper) sdk.Invariant {
 	return func(ctx sdk.Context) (string, bool) {
 		// msg, broke := RedemptionRateInvariant(k)(ctx)
 		// note: once we have >1 invariant here, follow the pattern from staking module invariants here: https://github.com/cosmos/cosmos-sdk/blob/v0.46.0/x/staking/keeper/invariants.go
-		return "", false
+		// return "", false
+		res, stop := k.BalanceStakeHostZoneInvariant()(ctx)
+		if !stop {
+			return res, stop
+		}
+
+		res, stop = k.ValidatorWeightHostZoneInvariant()(ctx)
+		if !stop {
+			return res, stop
+		}
+
+		return k.RedemptionRateInvariant()(ctx)
+	}
+}
+
+// BalanceStakeHostZoneInvariant ensure that balance stake of all host zone are equal to of validator's delegation
+func (k Keeper) BalanceStakeHostZoneInvariant() sdk.Invariant {
+	return func(ctx sdk.Context) (string, bool) {
+		listHostZone := k.GetAllHostZone(ctx)
+
+		for _, host := range listHostZone {
+			balanceStake := host.StakedBal
+			totalDelegateOfVals := k.GetTotalValidatorDelegations(host)
+			if !balanceStake.Equal(totalDelegateOfVals) {
+				return sdk.FormatInvariant(types.ModuleName, "balance-stake-hostzone-invariant", fmt.Sprintf(
+					"\tBalance stake of hostzone %s is not equal to total of validator's delegations \n"+
+						"\tBalance stake actually: %d\n"+
+						"\t Total of validator's delegations: %d\n",
+					host.ChainId, host.StakedBal, totalDelegateOfVals,
+				)), true
+			}
+		}
+		return sdk.FormatInvariant(types.ModuleName, "balance-stake-hostzone-invariant", "All host zones have balances stake is equal to total of validator's delegations"), false
+	}
+}
+
+// ValidatorWeightHostZoneInvariant ensure that sum of all validator delegation weight is equal to balance stake
+func (k Keeper) ValidatorWeightHostZoneInvariant() sdk.Invariant {
```

Comment:
I think this invariant will always fail when validator weights are changed, because there is a lag between the time a weight changes and the time delegations rebalance to match the new weight distribution. That would violate this invariant. For that reason, I think we should remove this invariant.